packages/twenty-front/src/modules/ui/input/components/Select.tsx

```diff
@@ -10,24 +11,31 @@ import { DropdownMenuSearchInput } from '@/ui/layout/dropdown/components/Dropdow
 import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
 import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
 import { MenuItem } from '@/ui/navigation/menu-item/components/MenuItem';
+import { MenuItemSelectAvatar } from '@/ui/navigation/menu-item/components/MenuItemSelectAvatar';
 
 import { SelectHotkeyScope } from '../types/SelectHotkeyScope';
 
-export type SelectOption<Value extends string | number | null> = {
+export type SelectOption<
+  Value extends string | number | null,
+  IconType extends IconComponent | React.ReactNode | null,
+> = {
   value: Value;
   label: string;
-  Icon?: IconComponent;
+  avatarOrIcon?: IconType;
```

Comment:
I know we had that in the past but maybe better to use the current pattern that's implemented on the codebase? For example on Chip I see `leftComponent`

Reply:
`Chip` renders the leftComponent as it's passed down from component props, adding that pattern on `Select`  would mean I'll have to pass in `MenuItemSelectAvatar` as a prop, I think it adding complexity which is not needed for the component, what do you think? :)

Reply:
I was about to say you should  thenprobably delete `MenuItemSelectAvatar` to only keep a refactored `MenuItemSelect` with a leftComponent, but it seems icons are already handle by `MenuItemSelectAvatar` ?
https://storybook.twenty.com/?path=/story/ui-navigation-menuitem-menuitemselectavatar--catalog

Reply:
You're right! I noticed it too, made changes to fit the `Select` component using `MenuItemSelectAvatar ` 👍